ding_dibs.module

```diff
@@ -420,3 +442,142 @@ function ding_dibs_cards() {
 
   return $cards;
 }
+
+/**
+ * Implements hook_cron().
+ * This is done by sending written reports and email adresses result cron launches in the database
+ */
+function ding_dibs_cron() {
+  $start_cron = FALSE;
+  $start = variable_get('ding_dibs_start_cron', '');
+  $start_minute = variable_get('ding_dibs_start_cron_minute', 0);
+  $start_operation = variable_get('ding_dibs_start_operation_cron', '');
+  $emails_list = variable_get('ding_dibs_emails', '');
+  if (!empty($start)) {
+    $start_val = mktime($start, $start_minute, 0, format_date(REQUEST_TIME, 'custom', 'm'), format_date(REQUEST_TIME, 'custom', 'd'), format_date(REQUEST_TIME, 'custom', 'Y'));
+    $today_date = format_date(REQUEST_TIME, 'custom', 'd/m/Y/H:i');
+    $start_date = format_date($start_val, 'custom', 'd/m/Y/H:i');
+    if (!empty($start_operation)) {
+      $start_operation_date = format_date($start_operation, 'custom', 'd/m/Y');
+    }
+    if ($today_date == $start_date || $today_date >= $start_date) {
+      if (empty($start_operation_date) || $start_operation_date < format_date(REQUEST_TIME, 'custom', 'd/m/Y')) {
+        $start_cron = TRUE;
+      }
+    }
+  }
+  if ($start_cron) {
+    if (!empty($emails_list)) {
+      $emails_array = preg_split("/\,|\r\n|\n/", $emails_list);
+      $emails_count = count($emails_array);
+      if ($emails_count >= 1) {
+        module_load_include('inc', 'ding_dibs', 'ding_dibs.email_reports');
+        $text_message = ding_dibs_email_reports();
+        variable_set('ding_dibs_start_operation_cron', REQUEST_TIME);
+        $id = _ding_dibs_insert_table(variable_get('ding_dibs_start_operation_cron', REQUEST_TIME));
```

Comment:
According to the developer, it has been implemented because cron stops working when array is processing. As a result, it will break a loop and there will be incomplete data. It helps to understand if a session is ragged and to identify a failed attempt of cron mailing.